pkg/services/notification/resource_control/task.go

```diff
@@ -63,36 +65,70 @@ func GetTask(taskId string) (*models.Task, error) {
 	return task, nil
 }
 
+func GetTasksByTaskIds(taskIds []string) ([]*models.Task, error) {
+	db := global.GetInstance().GetDB()
+	var tasks []*models.Task
+	err := db.Where("task_id in( ? )", taskIds).Find(&tasks).Error
+	if err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+func GetTasksByNfId(nfIds []string) ([]*models.Task, error) {
+	var tasks []*models.Task
+	err := nfdb.GetChain(global.GetInstance().GetDB().Where(models.TaskColNfId+" in (?)", nfIds).Find(&tasks)).Error
```

Comment:
No need to use GetChain. 
global.GetInstance().GetDB().Table(constants.TableTask).Where(models.TaskColNfId+" in (?)", nfIds).Find(&tasks)

Reply:
updated.